packages/bpk-component-datatable/src/sorter.js

```diff
@@ -0,0 +1,118 @@
+import _sortBy from 'lodash/sortBy';
+import { SortDirection } from 'react-virtualized';
+
+import { getSortIconDirection } from './bpkHeaderRenderer';
+
+const getSortDirection = (
+  sortProps,
+  newSortBy,
+  newSortDirection,
+  defaultSortDirection,
+) => {
+  const { sortBy, sortDirection } = sortProps;
+  if (newSortDirection !== null) {
+    return newSortDirection;
+  }
+  if (sortBy === newSortBy) {
+    return sortDirection === SortDirection.ASC
+      ? SortDirection.DESC
+      : SortDirection.ASC;
+  }
+  return defaultSortDirection;
+};
+
+class NaiveSorter {
+  constructor(props) {
+    const sortBy =
+      props.children.length > 0
+        ? props.children[props.defaultColumnSortIndex].props.dataKey
+        : undefined;
+    const sortDirection =
+      props.children[props.defaultColumnSortIndex].props.defaultSortDirection ||
+      SortDirection.ASC;
+    this.sortProps = {
+      sortBy,
+      sortDirection,
+    };
+    this.list = props.rows;
+    this.sort();
+  }
+
+  getRow = index => this.sortedList[index];
```

Comment:
This is subtley different from

```javascript
getRow(index) {
    return this.sortedList[index];
}
```

With the current way of writing this each instance of the class will delcare and allocate it's own copy of the function. With the second way of doing it each instance of the class will share the function via the class's [prototype chain](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain). 

There are a few cases where the style of declaration you've used is desirable, for example when the function needs to be bound to `this`, but in this instance we should prefer the second alternative for all these methods.

Reply:
Thanks, good to know about this. I just pushed a new commit with these changes